drivers/gpio/gpio_intel_socfpga.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_socfpga_gpio
+
+/**
+ * @file
+ * @brief Intel GPIO SoCFPGA Controller Driver
+ *
+ * The GPIO controller on Intel SoCFPGA serves
+ * as GPIO modules. This driver provides
+ * the GPIO functionality.
+ *
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_intel_socfpga);
+
+/* Check if reset property is defined */
+#define GPIO_INTEL_INST_RESET_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), resets)
+#define GPIO_INTEL_CHECK_RESET(inst)	GPIO_INTEL_INST_RESET_EN(inst) ||
+#define GPIO_INTEL_RESET_SUPPORT	(DT_INST_FOREACH_STATUS_OKAY(GPIO_INTEL_CHECK_RESET) 0)
+
+#if GPIO_INTEL_RESET_SUPPORT
```

Comment:
Please remove the conditionals around the include

Reply:
I gathered that. Can you please remove the conditionals around the include? It's something we try to avoid.

Reply:
^^ please make the suggested changes